Source/Timeline/AllDayView.swift

```diff
@@ -0,0 +1,163 @@
+
+import UIKit
+
+public protocol AllDayViewDataSource {
+  func numberOfAllDayEvents(in allDayView: AllDayView) -> Int
+  func allDayView(_ allDayView: AllDayView, eventDescriptorFor index: Int) -> EventDescriptor
+}
+
+public class AllDayView: UIView {
+  
+  let allDayLabelWidth: CGFloat = 53.0
+  let allDayEventHeight: CGFloat = 24.0
+  
+  public var dataSource: AllDayViewDataSource?
```

Comment:
While I recommend removing `AllDayViewDataSource` completely, there is an issue worth commenting.

If the `AllDayViewDataSource` owns  `AllDayView` (and this relationship is very common in iOS), there would be a retain cycle: AllDayView <-> AllDayViewDataSource.

To break it up, one of the properties is usually made weak:
```Swift
public weak var dataSource: AllDayViewDataSource?
```

Then, the relationship becomes strong in one direction and weak in the other:
```Swift
AllDayViewDataSource [i.e. the parent view] -----(strong)----> AllDayView
AllDayView  -   -  -   -   -   -     -   -   -  - (weak)  -  > AllDayViewDataSource
```

So, there would be no retain cycles.

Reply:
Darn.. It did slip my mind about the `TimeLineView` having a strong relationship to the `AllDayView` as well as the `AllDayView` to its `dataSource`

I'll update this